lib/smart_answer_flows/benefit-cap-calculator.rb

```diff
@@ -92,21 +99,52 @@ def define
           if response == "none"
             outcome :outcome_not_affected
           else
-            question BenefitCapCalculatorFlow.next_benefit_amount_question(config.questions, benefit_types)
+            question BenefitCapCalculatorFlow.next_benefit_amount_question(config.questions(chosen_cap), benefit_types)
+          end
+        end
+      end
+
+      #Q4 future flow
+      checkbox_question :receiving_non_exemption_benefits_future? do
+        config.benefits(:future).keys.each do |benefit|
+          option benefit
+        end
+
+        next_node_calculation :benefit_types do |response|
+          response.split(",").map(&:to_sym)
+        end
+
+        precalculate :benefit_options do
+          config.descriptions(chosen_cap).merge(none_above: "None of the above")
+        end
+
+        calculate :total_benefits do
+          0
+        end
+
+        calculate :benefit_cap do
+          0
+        end
```

Comment:
I think the `benefit_options`, `total_benefits` & `benefit_cap` blocks are common to both versions of Q4 and since we always arrive at these via Q3, I think these blocks could move there. Having said that I would do that in a separate commit i.e. introduce the duplication first and then remove it in a subsequent commit. Note also that `benefit_options` should become a `calculate` block if you do move them.